core/src/main/scala/zio/schema/codec/ProtobufCodec.scala

```diff
@@ -383,6 +384,7 @@ object ProtobufCodec extends Codec {
         case Schema.Optional(codec)           => optionalDecoder(codec).asInstanceOf[Decoder[A]]
         case Schema.Fail(message)             => (_, _) => Left(message)
         case Schema.EitherSchema(left, right) => eitherDecoder(left, right).asInstanceOf[Decoder[A]]
+        case _                                => ???
```

Comment:
Quick fix here and elsewhere will be `case caseClass : CaseClass[A] => decoder(caseClass.toRecord)`

Reply:
This won't work because you end up with a `Decoder[Map[String,_]]` instead of a `Decoder[A]`

I think it shouldn't be too hard to implement the protobuf encoder/decoder for the case class schemas though.